vernac/declare.ml

```diff
@@ -63,6 +70,7 @@ module CInfo = struct
 
 
   let make ~name ~typ ?(args=[]) ?(impargs=[]) ?using () =
+    check_exists name;
```

Comment:
Uh uh , `CInfo` I don't like CInfo.make getting a dependency on `Global`, this seems very brittle. The check should be added at the right place, which seems it is not here.

Reply:
Yeah it's probably too low level.

Reply:
Yes this info is passed around and stored so an imperative check on build makes no sense.